src/DataBus/AzureDataBusPersistence.cs

```diff
@@ -21,64 +17,43 @@ protected override void Setup(FeatureConfigurationContext context)
+                    MaxRetries = dataBusSettings.MaxRetries,
+                    Mode = RetryMode.Fixed
+                }
+            };
+            return new BlobContainerClient(dataBusSettings.ConnectionString, dataBusSettings.Container, clientOptions);
         }
 
-        static CloudBlobContainer CreateCloudBlobContainer(DataBusSettings dataBusSettings)
+        private void ThrowIfMissingConfigurationForBlobContainer(bool isBlobClientConfiguredByUser,
+            DataBusSettings dataBusSettings)
         {
-            CloudBlobContainer container;
-
-            // Attempt managed identity identity first
-            if (!string.IsNullOrWhiteSpace(dataBusSettings.StorageAccountName))
+            if (isBlobClientConfiguredByUser)
             {
-                var azureServiceTokenProvider = new AzureServiceTokenProvider();
-                var state = (azureServiceTokenProvider, dataBusSettings);
-                var tokenAndFrequency = TokenRenewerAsync(state, CancellationToken.None).GetAwaiter().GetResult();
-                var tokenCredential = new TokenCredential(tokenAndFrequency.Token, TokenRenewerAsync, state, tokenAndFrequency.Frequency.Value);
-                var storageCredentials = new StorageCredentials(tokenCredential);
-                var containerPath = $"https://{dataBusSettings.StorageAccountName}.blob.{dataBusSettings.EndpointSuffix}/{dataBusSettings.Container}";
-
-                container = new CloudBlobContainer(new StorageUri(new Uri(containerPath)), storageCredentials);
+                return;
             }
-            else // fallback to connection string
-            {
-                var cloudBlobClient = CloudStorageAccount.Parse(dataBusSettings.ConnectionString).CreateCloudBlobClient();
-                container = cloudBlobClient.GetContainerReference(dataBusSettings.Container);
-            }
-
-            return container;
-        }
 
-        static async Task<NewTokenAndFrequency> TokenRenewerAsync(object state, CancellationToken token = default)
-        {
-            var (azureServiceTokenProvider, settings) = (ValueTuple<AzureServiceTokenProvider, DataBusSettings>)state;
-
-            // Use the same token provider to request a new token.
-            var resourceUri = $"https://{settings.StorageAccountName}.blob.{settings.EndpointSuffix}";
-            var result = await azureServiceTokenProvider.GetAuthenticationResultAsync(resourceUri, cancellationToken: token).ConfigureAwait(false);
-
-            // Renew the token before it expires.
-            var next = (result.ExpiresOn - DateTimeOffset.UtcNow) - settings.RenewalTimeBeforeTokenExpires;
-            if (next.Ticks < 0)
+            if (!dataBusSettings.UserProvidedConnectionString)
             {
-                next = default;
+                throw new Exception("Unable to find a configured BlobClient in the container and unable to fall back to a connectionstring + container as none were supplied.");
```

Comment:
Are we trying to resolve a BlobClient from a container? Or from the settings, meaning the message is somewhat misleading and should state that neither a BlobClient nor a connection string was provided.

```suggestion
                throw new Exception("Azure databus was not configured to use a BlobContainerClient. Use '.BlobContainerClient()' to provide a BlobContainerClient to be used. Alternatively, configure the data bus using a connection string and a container name.");
```